packages/sync-engine/src/stripeSync.ts

```diff
@@ -886,6 +929,38 @@ export class StripeSync {
     return this.postgresClient.upsertMany(creditNotes, 'credit_notes', creditNoteSchema)
   }
 
+  private async upsertCheckoutSessions(
+    checkoutSessions: Stripe.Checkout.Session[],
+    backfillRelatedEntities?: boolean
+  ): Promise<Stripe.Checkout.Session[]> {
+    if (backfillRelatedEntities ?? this.config.backfillRelatedEntities) {
+      await Promise.all([
+        this.backfillCustomers(getUniqueIds(checkoutSessions, 'customer')),
+        this.backfillSubscriptions(getUniqueIds(checkoutSessions, 'subscription')),
+        this.backfillPaymentIntents(getUniqueIds(checkoutSessions, 'payment_intent')),
+        this.backfillInvoices(getUniqueIds(checkoutSessions, 'invoice')),
+      ])
+    }
+
+    // Upsert checkout sessions first
+    const rows = await this.postgresClient.upsertMany(
+      checkoutSessions,
+      'checkout_sessions',
+      checkoutSessionSchema
+    )
+
+    // Upsert line items into separate table
+    const lineItemsPromises = checkoutSessions
```

Comment:
This should be a single upsert and not one upsert per line item

Are line items sent via the webhook by default? Otherwise the checkout session line items would never be there for a regular webhook